server/utils/pdfRenderer.ts

```diff
@@ -30,15 +50,16 @@ export function pdfRenderer(client: GotenbergClient) {
               throw bodyError
             }
 
-            res.header('Content-Type', 'application/pdf')
-            res.header('Content-Transfer-Encoding', 'binary')
-            res.header('Content-Disposition', `inline; filename=${options.filename}`)
-
             client
-              .renderPdfFromHtml(pageHtml, headerHtml, footerHtml, options?.pdfMargins)
-              .then(buffer => res.send(buffer))
-              .catch(reason => {
-                logger.warn(reason)
+              .renderPdfFromHtml(pageHtml, headerHtml, footerHtml, readAssetCss(), options?.pdfMargins)
+              .then(buffer => {
+                res.header('Content-Type', 'application/pdf')
```

Comment:
only do the header writing if we've rendered the pdf.  This then allows the error page to be rendered if the pdf generation fails